### ISSUE TITLE
Gem failing with edge versions of it's dependencies

### ISSUE THREAD

**Issue by mcelicalderon** [graphql-devise/graphql_devise#42]
We need to fix the currently allowed failures for the edge version of the Gemfile

**Comment by aarona** [comment]
I did some research on this tonight. It looks like edge Rails is causing this and graphql edge probably isn't. I didn't find a solution to this yet but the following was my train of thought while trying to solve this.

First of all, if you guys didn't know this trick (I had to look up how to do this) you can reference rails edge by doing this, which will allow you to reproduce the errors:

**Gemfile:**
```ruby
source 'https://rubygems.org'

gem 'rails', github: "rails/rails"

gemspec
```

**graphql_devise.gemspec:**
```ruby
...
spec.add_dependency 'rails' #, '>= 4.2'
...
```

Here's the odd thing: When you run:
```
prompt> rspec .
```
You'll notice you'll get an error for testing the version number:
```
GraphqlDevise
  has a version number (FAILED - 1)
....etc...
Failures:

  1) GraphqlDevise has a version number
     Failure/Error:
       raise WrongScopeError,
             "`#{name}` is not available from within an example (e.g. an " \
             "`it` block) or from constructs that run in the scope of an " \
             "example (e.g. `before`, `let`, etc). It is only available " \
             "on an example group (e.g. a `describe` or `context` block)."

       `name` is not available from within an example (e.g. an `it` block) or from constructs that run in the scope of an example (e.g. `before`, `let`, etc). It is only available on an example group (e.g. a `describe` or `context` block).

```

but when you run just that one test like so, it works:
```
prompt> rspec spec/graphql_devise_spec.rb                    

GraphqlDevise
  has a version number

Finished in 0.00342 seconds (files took 0.07614 seconds to load)
1 example, 0 failures
```
This isn't the case for other specs however. I did some research on the `WrongScopeError` message and  the `#{name}` variable changes from `name` to `expect` when you change `spec/graphql_devise_spec.rb` to:

```ruby
RSpec.describe GraphqlDevise do
  expect(GraphqlDevise::VERSION).not_to be_nil
  it 'has a version number' do
  end
end
```
Run the spec:
```
prompt> rspec spec/graphql_devise_spec.rb

An error occurred while loading ./spec/graphql_devise_spec.rb.
Failure/Error: expect(GraphqlDevise::VERSION).not_to be_nil
  `expect` is not available on an example group (e.g. a `describe` or `context` block). It is only available from within individual examples (e.g. `it` blocks) or from constructs that run in the scope of an example (e.g. `before`, `let`, etc).
# ./spec/graphql_devise_spec.rb:2:in `block in <top (required)>'
# ./spec/graphql_devise_spec.rb:1:in `<top (required)>'
No examples found.

Finished in 0.00003 seconds (files took 0.09383 seconds to load)
0 examples, 0 failures, 1 error occurred outside of examples
```

Seems like something going on with RSpec and not really graphql_devise code itself. If you check out this SO post: https://stackoverflow.com/questions/30135038/rspec-wrongscopeerror-expect-is-not-available-on-an-example-group it looks like this is definitely RSpec complaining and not rails. `WrongScopeError` is an RSpec exception it looks like.

Then I thought that maybe if I'm using rails edge, I should be using rspec-rails edge also but it requires `rspec-core (= 3.10.0.pre)" and that gem isn't in any of the sources:

```
prompt> bundle                          
The git source `git://github.com/rails/rails.git` uses the `git` protocol, which transmits data without encryption. Disable this warning with `bundle config git.allow_insecure true`, or switch to the `https` protocol to keep your data secure.
The git source `git://github.com/rspec/rspec-rails.git` uses the `git` protocol, which transmits data without encryption. Disable this warning with `bundle config git.allow_insecure true`, or switch to the `https` protocol to keep your data secure.
Fetching git://github.com/rspec/rspec-rails.git
Fetching gem metadata from https://rubygems.org/.........
Fetching gem metadata from https://rubygems.org/.
Could not find gem 'rspec-core (= 3.10.0.pre)' in any of the gem sources listed in your Gemfile.
```

If you look at rspec-rails's releases page: https://github.com/rspec/rspec-rails/releases you'll see that they did a hotfix and bumped 3.9.0 down to 3.8.3 only a day later.

**TL;DR**: I think there is some volatility going on with bleeding edge RSpec possibly? We are using version 3.9.0 but then again, I switched to the latest stable release of Rails (6.0.1) and the tests passed.

**Comment by aarona** [comment]
Ok I looked some more at rspec-rails and what I thought was a hot fix was really just a version bump change from 3.8.2 -> 3.9.0 to 3.8.2 -> 3.8.3. Probably they decided the next version was going to be a major change and ditched a 3.9.0 because they're moving to 4.0.0 beta.

This is definitely Rails edge as Rails 6.0.1 is behaving.

**Comment by mcelicalderon** [comment]
Nice, thanks for looking into this. I'll try to push a fix ASAP

**Comment by mcelicalderon** [comment]
Closing this one as this is a problem only with the latest (master) rails. mIght be lack of compatibility with RSpec from a quick test. Anyway, not really related to this gem.

**Comment by davidrunger** [comment]
@aarona I think that this issue
```
`name` is not available from within an example (e.g. an `it` block) [...]
```

might have been introduced by https://github.com/rails/rails/pull/37770 ?

That PR switched a method call from `method_name` to `name` ([here](https://github.com/rails/rails/pull/37770/files#diff-793053363f19bfca117c441bbb685b83R101)). I think that `name` is sometimes not defined, which gets bubbled up through some `method_missing`s in RSpec and produces this error, I think.

At least, that's what is happening in my app when I try to move to `rails` installed from `master`, and then Googling for the above error message brought me here.

I'm working around the issue for now with a monkeypatch to go back to using `method_name`:
```rb
# config/initializers/monkeypatches.rb
#
# This fixes what seems to be a bug introduced by
# https://github.com/rails/rails/pull/37770
# "Modify ActiveRecord::TestFixtures to not rely on AS::TestCase:"
#
module ActiveRecord::TestFixtures
  def run_in_transaction?
    use_transactional_tests &&
      !self.class.uses_transaction?(method_name) # this monkeypatch changes `name` to `method_name`
  end
end
```

**Comment by aarona** [comment]
Thanks for the information, @davidrunger !

**Comment by mcelicalderon** [comment]
Happy to confirm this is fixed with the recent release of RSpec 4.0! Thank you both for looking into this. @davidrunger  you can get rid of the monkey patch now 🙌. #72 takes care of this.

**Comment by epadillac** [comment]
> @aarona I think that this issue
> 
> ```
> `name` is not available from within an example (e.g. an `it` block) [...]
> ```
> 
> might have been introduced by [rails/rails#37770](https://github.com/rails/rails/pull/37770) ?
> 
> That PR switched a method call from `method_name` to `name` ([here](https://github.com/rails/rails/pull/37770/files#diff-793053363f19bfca117c441bbb685b83R101)). I think that `name` is sometimes not defined, which gets bubbled up through some `method_missing`s in RSpec and produces this error, I think.
> 
> At least, that's what is happening in my app when I try to move to `rails` installed from `master`, and then Googling for the above error message brought me here.
> 
> I'm working around the issue for now with a monkeypatch to go back to using `method_name`:
> 
> ```ruby
> # config/initializers/monkeypatches.rb
> #
> # This fixes what seems to be a bug introduced by
> # https://github.com/rails/rails/pull/37770
> # "Modify ActiveRecord::TestFixtures to not rely on AS::TestCase:"
> #
> module ActiveRecord::TestFixtures
>   def run_in_transaction?
>     use_transactional_tests &&
>       !self.class.uses_transaction?(method_name) # this monkeypatch changes `name` to `method_name`
>   end
> end
> ```

Thanks @davidrunger, this saved my life

**Comment by RaminMammadzada** [comment]
> @aarona I think that this issue
> 
> ```
> `name` is not available from within an example (e.g. an `it` block) [...]
> ```
> 
> might have been introduced by [rails/rails#37770](https://github.com/rails/rails/pull/37770) ?
> 
> That PR switched a method call from `method_name` to `name` ([here](https://github.com/rails/rails/pull/37770/files#diff-793053363f19bfca117c441bbb685b83R101)). I think that `name` is sometimes not defined, which gets bubbled up through some `method_missing`s in RSpec and produces this error, I think.
> 
> At least, that's what is happening in my app when I try to move to `rails` installed from `master`, and then Googling for the above error message brought me here.
> 
> I'm working around the issue for now with a monkeypatch to go back to using `method_name`:
> 
> ```ruby
> # config/initializers/monkeypatches.rb
> #
> # This fixes what seems to be a bug introduced by
> # https://github.com/rails/rails/pull/37770
> # "Modify ActiveRecord::TestFixtures to not rely on AS::TestCase:"
> #
> module ActiveRecord::TestFixtures
>   def run_in_transaction?
>     use_transactional_tests &&
>       !self.class.uses_transaction?(method_name) # this monkeypatch changes `name` to `method_name`
>   end
> end
> ```

This also helped to solve the problem.
